### PR TITLE
Use NumPy to compute log2 as LongTensor does not support it

### PR DESCRIPTION
**Pull request by mccorby**
Fixed issue when calculating whether to use LPT or FPT for LongTensor tensors.

Issue: https://github.com/OpenMined/PySyft/issues/2327

### CODE REVIEW

File: test/torch/tensors/test_large_precision.py [hunk 1]
```diff
@@ -218,3 +218,12 @@ def test_mod():
     lpt2 = x2.fix_prec(internal_type=internal_type, precision_fractional=precision_fractional)
     result = lpt1 % lpt2
     assert torch.all(torch.eq(expected, result.float_precision()))
+
+
+def test_long():
```
Comment (mari-linhares) [inline test/torch/tensors/test_large_precision.py]: can you parametrize this test and use multiple more input + expected values? Like an array with more than one value, float numbers with more decimal places (example 1.234123), negative numbers...

File: test/torch/tensors/test_precision.py [hunk 2]
```diff
@@ -18,6 +18,15 @@ def test_wrap(workers):
     assert isinstance(x.child.child, torch.Tensor)
 
 
+def test_long():
```
Comment (mari-linhares) [inline test/torch/tensors/test_precision.py]: same

File: test/torch/tensors/test_large_precision.py [hunk 3]
```diff
@@ -218,3 +218,19 @@ def test_mod():
     lpt2 = x2.fix_prec(internal_type=internal_type, precision_fractional=precision_fractional)
     result = lpt1 % lpt2
     assert torch.all(torch.eq(expected, result.float_precision()))
+
+
+test_data = [
```
Comment (mari-linhares) [inline test/torch/tensors/test_large_precision.py]: I don't think is needed to create a variable just to use it in the decorator. Also, can you add tests with more decimal places (I assume this would work shouldn't it?) and negative numbers?

Comment (mccorby) [reply test/torch/tensors/test_large_precision.py]: Thanks for the reviews!

"I don't think is needed to create a variable just to use it in the decorator"
What would you suggest instead? Should I loop inside the test itself?

Comment (mari-linhares) [reply test/torch/tensors/test_large_precision.py]: You can move this array to the decorator itself. You'll have something like:

```python
 @pytest.mark.parametrize("x, expected", [(torch.tensor([1]), torch.tensor([1.0])),
    (torch.tensor([1.0]), torch.tensor([1.0])),
    (torch.tensor([2000.0, 1.0]), torch.tensor([2000.0, 1.0])),
    (torch.tensor([2000.0, 1]), torch.tensor([2000.0, 1.0])),
])
```

Comment (mccorby) [reply test/torch/tensors/test_large_precision.py]: Done

File: test/torch/tensors/test_large_precision.py [hunk 4]
```diff
@@ -218,3 +219,22 @@ def test_mod():
...
+@pytest.mark.parametrize(
+    "x, expected",
+    [
+        (torch.tensor([1]), torch.tensor([1.0])),
+        (torch.tensor([1.0]), torch.tensor([1.0])),
+        (torch.tensor([2000.0, 1.0]), torch.tensor([2000.0, 1.0])),
+        (torch.tensor([2000.0, 1]), torch.tensor([2000.0, 1.0])),
+        (torch.tensor([-2000.0]), torch.tensor([-2000.0])),
+        (torch.tensor([-2000.0, -50]), torch.tensor([-2000.0, -50.0])),
+        (torch.tensor([-2000.0123458910]), torch.tensor([-2000.0123458910])),
```
Comment (robert-wagner) [inline test/torch/tensors/test_large_precision.py]: Could you add a test case where there are both positive and negative values in a tensor and one with a positive tensor with values after the decimal place

**Comment by robert-wagner** [comment]
@mccorby please run black on the code

File: test/torch/tensors/test_large_precision.py (hunk 4 above)
Comment (mccorby) [reply test/torch/tensors/test_large_precision.py]: Done
